daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetInputter.scala

```diff
@@ -413,14 +416,23 @@ abstract class InfosetInputter
         } catch {
           case ex: NonTextFoundInSimpleContentException =>
             UnparseError(One(elem.erd.schemaFileLocation), Nope, ex.getMessage())
+          case uri: URISyntaxException =>
```

Comment:
I wonder why we haven't seen this before? Maybe one of your new tests found this?

This feels odd because ``getSimpleText`` doesn't do any conversion to the primitive type, that I can see. It just gets a string value from the XML/Json/whatever infoset. Any idea where this is coming from?

Reply:
So this came from the new blob_unparseError test, which ends up calling getSimpleText in TDMLInfosetInputter, which has a special case for xs:anyURI. If the primType is AnyURI, it simply creates a new URI object and has no try/catch for invalid URI values.

Reply:
Ah, this feel like an issue in the TDMLInfosetInputter. The getSimpleText documentation says:
```
  /**
   * Get the content of a simple type. This will only be called when the
   * current event type is StartElement and the element is a simple type. If
   * the event contains complex data, it is an error and should throw
   * NonTextFoundInSimpleContentException. If the element does not have any
   * simple content, this should return either null or the empty string.
   */
```
So I think it's wrong for getSimpleText to throw an exception other than NonTextFoundInSimpleContentException. The TDMLInfosetInputter should probably just return ``res`` if it's not a value URI and let daffodil eventually figure out that it's not valid like it does with other primitive types.

Reply:
Missed this before pushing up my last commit, will fix first thing tomorrow.